cni/pkg/nodeagent/server.go

```diff
@@ -235,91 +239,107 @@ func (s *meshDataplane) Stop(skipCleanup bool) {
 	s.netServer.Stop(skipCleanup)
 }
 
+// AddPodToMesh attempts to inject iptables rules into the pod, and sends the pod to ztunnel.
+// If this succeeds, it will add the pod to the node ipset, and finally annotate the pod
+// (indicating it has been mutated/captured) with a captured status.
+//
+// If anything fails *before* sending the pod to ztunnel, returns a nonrecoverable/nonretryable error.
```

Comment:
FWIW the CNI plugin is going to retry these regardless of if we mark them retryable or not. Should be fine, but maybe a bit confusing

Reply:
Right, the CNI plugin effectively has its own retry cycle outside of the informer, the "retryable" error is moot for the plugin (it doesn't care).

To your point, better error name might help.